src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -1222,13 +1222,20 @@ void program_node::init_onednn_primitive_attributes() {
     if (fused_ops.size() > 1) {
         dnnl::post_ops optimized_post_ops = post_ops;
         bool optimization_is_finished = false;
+        int trial_count = 0;
 
         add_onednn_fused_primitives(fused_ops);
 
         // Trying to combine multiplications and additions which are placed one after another.
         // We do it in the cycle because some optimization cases can be simplified again from time to time
         do {
+            GPU_DEBUG_GET_INSTANCE(debug_config);
+            GPU_DEBUG_IF(debug_config->disable_onednn_opt_post_ops)
+                break;
             optimized_post_ops = try_optimize_post_ops(optimized_post_ops, attrs, optimization_is_finished);
+            trial_count++;
+            if (trial_count == 100)
```

Comment:
Let's add some macro for defining max optimization attempts

Reply:
Will be removed.